TTS/tts/models/vits.py

```diff
@@ -396,13 +555,18 @@ def forward(
             - speaker_ids: :math:`[B]`
         """
         outputs = {}
-        sid, g = self._set_cond_input(aux_input)
-        x, m_p, logs_p, x_mask = self.text_encoder(x, x_lengths)
-
+        sid, g, lid = self._set_cond_input(aux_input)
         # speaker embedding
-        if self.num_speakers > 1 and sid is not None:
+        if self.args.use_speaker_embedding and sid is not None and not self.use_d_vector:
```

Comment:
`self.args.use_speaker_embedding and sid is not None` should be enough ?

Reply:
Indeed